tests/model_serving/model_server/private_endpoint/conftest.py

```diff
@@ -0,0 +1,195 @@
+    endpoint_namespace: Namespace,
+    endpoint_s3_secret: Secret,
+    aws_access_key_id: str,
+    aws_secret_access_key: str,
+    s3_bucket_name_wisdom: str,
+) -> Generator[Secret, None, None]:
+    secret = {
+        "access_key_id": aws_access_key_id,
+        "bucket": s3_bucket_name_wisdom,
+        "default_bucket": s3_bucket_name_wisdom,
+        "endpoint_url": AWS_ENDPOINT_EAST_2,
+        "region": AWS_REGION_EAST_2,
+        "secret_access_key": aws_secret_access_key,
+        "type": "s3",
+    }
+    data = {"endpoint-s3-secret": b64_encoded_string(json.dumps(secret))}
+    with Secret(
+        client=admin_client,
+        namespace=endpoint_namespace.name,
+        data_dict=data,
+        wait_for_resource=True,
+        name="storage-config",
+    ) as storage_config:
+        yield storage_config
+
+
+@pytest.fixture()
+def service_mesh_member(
+    admin_client: DynamicClient, diff_namespace: Namespace
+) -> Generator[ServiceMeshMember, None, None]:
+    with ServiceMeshMember(
+        client=admin_client,
+        namespace=diff_namespace.name,
+        name="default",
+        control_plane_ref={"name": "data-science-smcp", "namespace": "istio-system"},
+        wait_for_resource=True,
+    ) as smm:
+        yield smm
+
+
+@pytest.fixture()
+def endpoint_pod_with_istio_sidecar(admin_client: DynamicClient, endpoint_namespace: Namespace) -> Pod:
+    pod = create_sidecar_pod(
+        admin_client=admin_client,
+        namespace=endpoint_namespace.name,
+        istio=True,
+        pod_name="test-with-istio",
+    )
+    yield pod
+    pod.clean_up()
+
+
+@pytest.fixture()
+def endpoint_pod_without_istio_sidecar(admin_client: DynamicClient, endpoint_namespace: Namespace) -> Pod:
+    pod = create_sidecar_pod(
+        admin_client=admin_client,
+        namespace=endpoint_namespace.name,
+        istio=False,
+        pod_name="test",
+    )
+    yield pod
+    pod.clean_up()
+
+
+@pytest.fixture()
+def diff_pod_with_istio_sidecar(admin_client: DynamicClient, diff_namespace: Namespace) -> Pod:
+    pod = create_sidecar_pod(
+        admin_client=admin_client,
+        namespace=diff_namespace.name,
+        istio=True,
+        pod_name="test-with-istio",
+    )
+    yield pod
+    pod.clean_up()
+
+
+@pytest.fixture()
+def diff_pod_without_istio_sidecar(admin_client: DynamicClient, diff_namespace: Namespace) -> Pod:
+    pod = create_sidecar_pod(
+        admin_client=admin_client,
+        namespace=diff_namespace.name,
+        istio=False,
+        pod_name="test",
+    )
+    yield pod
+    pod.clean_up()
+
+
+@pytest.fixture()
+def running_flan_pod(admin_client: DynamicClient, endpoint_isvc: InferenceService) -> None:
+    predictor_pod = get_flan_pod(
+        namespace=endpoint_isvc.namespace,
+        client=admin_client,
+        name_prefix=endpoint_isvc.name,
+    )
+    predictor_pod.wait_for_status(status="Running")
+    predictor_pod.wait_for_condition(condition="Ready", status="True")
```

Comment:
why wait for both?

Reply:
See https://github.com/opendatahub-io/opendatahub-tests/pull/22/files/38b1f728c8d82955805212d36cf8c47f03d17a10#r1846653516